kore/src/Kore/Step/Function/Evaluator.hs

```diff
@@ -141,15 +141,21 @@ evaluateApplication
     finishT = exceptT return return
 
     Application { applicationSymbolOrAlias = symbol } = application
+    Application { applicationChildren = children } = application
     Symbol { symbolAttributes } = symbol
 
     termLike = synthesize (ApplySymbolF application)
     unevaluated =
         OrPattern.fromPattern
         $ Pattern.withCondition
-            (TermLike.markSimplified termLike)
+            (markSimplifiedIfChildren termLike)
             childrenPredicate
 
+    markSimplifiedIfChildren =
+        if all TermLike.isSimplified children
```

Comment:
`Application` is `Foldable`, so you don't need to explicitly unpack the `children` here:
```suggestion
        if all TermLike.isSimplified application
```